### PR TITLE
fix forgot removed records when deleting user

### LINKED ISSUE #5356: GPG-Keys don't get deleted when deleting an account
- Gitea version: 8733384
- Git version: Not relevant
- Operating system: Alpine Linux 3.7
- Database:
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes
  - [ ] No
  - [ ] Not relevant

## Description
When you delete a profile with an existing GPG-Key, you can't use this GPG-Key in a new account again. An error says "A public GPG key with same ID already exists.", and it can't be added.
Deleting a Key alone (not the whole account) and adding it again manually is no problem; My guess is that in the process of the account deletion, the GPG-Keys don't get deleted as well.

### PR DESCRIPTION
**Pull request by lunny**
Will fix #5356 and replace #5357 , also update xorm to latest stable version.

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

**Comment by lunny** [comment]
@lafriks yes, will fix that.

**Comment by kolaente** [comment]
Conflicts.

File: models/user.go [hunk 1]
```diff
@@ -1015,25 +1015,26 @@ func deleteUser(e *xorm.Session, u *User) error {
...
-	keyIDs := make([]int64, len(keys))
-	for i := range keys {
-		keyIDs[i] = keys[i].ID
-	}
-	if err = deletePublicKeys(e, keyIDs...); err != nil {
+	if _, err = e.Delete(&PublicKey{OwnerID: u.ID}); err != nil {
 		return fmt.Errorf("deletePublicKeys: %v", err)
 	}
+	rewriteAllPublicKeys(e)
 	// ***** END: PublicKey *****
```
Comment (zeripath) [inline models/user.go:1030]: RewriteAllPublicKeys is called at the end of DeleteUser too, does it still need to be called there?

File: models/user.go [hunk 2]
```diff
@@ -1087,6 +1088,7 @@ func DeleteUser(u *User) (err error) {
 	if err = sess.Commit(); err != nil {
 		return err
 	}
+	sess.Close()
 
 	return RewriteAllPublicKeys()
```
Comment (zeripath) [inline models/user.go]: See above

**Comment by lunny** [comment]
@zeripath Good catch. Done.

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
@lunny Does this need backporting?

**Comment by lunny** [comment]
@zeripath #5509 is the backport of this one.

**Pull request merged**